tests/unit/tls_test.cc

```diff
@@ -1444,6 +1444,10 @@ SEASTAR_THREAD_TEST_CASE(test_dn_name_handling) {
         fout.get();
 
         auto dn = fdn.get();
+        BOOST_REQUIRE(dn.has_value());
```

Comment:
Are there tests for the failure modes?

Reply:
Nope, i believe these unit tests could be enhanced a bit more